File: frontend/lib/models/reaction.dart
import 'constants.dart';

class Reaction {
  String reactionId;
  String userId;
  String contentId;
  ReactionType reactionType;
  DateTime timestamp;

  Reaction(
    this.reactionId,
    this.userId,
    this.contentId,
    this.reactionType,
    this.timestamp,
  );

  (String, ReactionType, DateTime) getUserReactions() {
    return (contentId, reactionType, timestamp);
  }
}

File: backend/personalization/models/reaction.py
from enum import Enum
from dataclasses import dataclass
from datetime import datetime
from typing import List, Tuple

class ReactionType(Enum):
    NotFunny = "NotFunny"
    Meh = "Meh"
    Funny = "Funny"
    Hilarious = "Hilarious"
    # You can easily extend this enum as the app evolves.

@dataclass
class Reaction:
    reaction_id: str 
    user_id: str
    content_id: str
    reaction_type: ReactionType
    timestamp: datetime

    def get_user_reaction(self) -> Tuple[str, ReactionType, datetime]:
        return (self.content_id, self.reaction_type, self.timestamp)
